### PR TITLE
[Function] add BE bitmap function bitmap_subset_in_range

### LINKED ISSUE #6875: [Feature] I hope to support more bitmap functions with the same functions as Clickhouse
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

1. bitmapSubsetInRange

bitmapSubsetInRange(bitmap, range_start, range_end)
参数：
bitmap – 位图对象.
range_start – 范围起始点（含）.
range_end – 范围结束点（不含）

2. bitmapSubsetLimit

bitmapSubsetLimit(bitmap, range_start, limit)
参数：
bitmap – 位图对象.
range_start – 范围起始点（含）.
limit – 子位图基数上限

3. subBitmap
subBitmap(bitmap, offset, limit)
参数：
bitmap – 位图对象.
offset – 跳过多少个元素.
limit – 子位图基数上限

### Use case

需求：
我们有一个系统是基于用户画像的推送系统，根据用户画像的标签构建用户群组bitmap，然后根据构建的bitmap导出用户id，
基于用户id去做推送。
遇到的问题：
目前遇到的问题是，当构建的bitmap数据量比较大时（比如达到百万级别的基数），我们无法通过bitmap_to_string函数导出
所有的用户id，所以我们想的办法是通过分页的方式，每次构建子bitmap，然后将子bitmap进行导出

### Related issues

#bit

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

Add bitmap function `bitmap_subset_in_range`.
This function will return subset in specified range (not include the range_end).

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6875 ) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/util/bitmap_value.h [hunk 1]
```diff
@@ -1451,6 +1451,26 @@ class BitmapValue {
...
+                continue;
+            }
+            if (*it < range_end) {
+                ret_bitmap->add(*it);
+                ++count;
+            } else {
+                break;
+            }
+        }
+        return count;
```
Comment (xinyiZzz) [inline be/src/util/bitmap_value.h:1470]: If range_start is greater than bitmap max, count will be equal to 0, and finally an empty string will be output, whether to unified into "null"

Comment (qidaye) [reply be/src/util/bitmap_value.h:1470]: When there is a error, it returns `null`. The case `range_start` greater than `bitmap_max` is a normal case, I think it's ok to return an empty result.

**Review by morningman — APPROVED** [review]
LGTM

File: be/test/exprs/bitmap_function_test.cpp [hunk 2]
```diff
@@ -552,6 +552,67 @@ TEST_F(BitmapFunctionsTest, bitmap_max) {
...
+
+    res = BitmapFunctions::bitmap_subset_in_range(ctx, bitmap_src, BigIntVal(0), BigIntVal(1));
+    result = BitmapFunctions::bitmap_count(ctx, res);
+    ASSERT_EQ(BigIntVal(1), result);
+
+    res = BitmapFunctions::bitmap_subset_in_range(ctx, bitmap_src, BigIntVal(11), BigIntVal(15));
+    result = BitmapFunctions::bitmap_count(ctx, res);
+    ASSERT_EQ(BigIntVal(4), result);
+
+    res = BitmapFunctions::bitmap_subset_in_range(ctx, bitmap_src, BigIntVal(11), DecimalV2Value::MAX_INT64);
```
Comment (xinyiZzz) [inline be/test/exprs/bitmap_function_test.cpp:583]: It seems to be better to use BigIntVal(INT64_MAX) and BigIntVal(INT64_MIN) here

File: be/test/exprs/bitmap_function_test.cpp [hunk 3]
```diff
@@ -552,6 +552,67 @@ TEST_F(BitmapFunctionsTest, bitmap_max) {
...
+
+    // empty
+    BitmapValue bitmap0;
+    StringVal empty_str = convert_bitmap_to_string(ctx, bitmap0);
+    res = BitmapFunctions::bitmap_subset_in_range(ctx, empty_str, BigIntVal(1), BigIntVal(3));
+    BigIntVal result = BitmapFunctions::bitmap_count(ctx, res);
+    ASSERT_EQ(BigIntVal(0), result);
+
+    // normal
+    BitmapValue bitmap1({0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,208,23,24,25,26,27,28,29,30,31,32,33,100,200,500});
```
Comment (xinyiZzz) [inline be/test/exprs/bitmap_function_test.cpp:568]: 208,

**Pull request merged**